src/db.c

```diff
@@ -2145,9 +2153,14 @@ int expandDb(const redisDb *db, uint64_t db_size, dbKeyType keyType) {
                 } else {
                     d = db->expires[i];
                 }
-                if (dictTryExpand(d, (db_size / server.cluster->myself->numslots)) != DICT_OK) {
-                    serverLog(LL_WARNING, "Trying to expand dict of slot number %d", i );
+                int result = try_expand ? dictTryExpand(d, db_size) : dictExpand(d, db_size);
+                if (try_expand && result == DICT_ERR) {
+                    serverLog(LL_WARNING, "Dict expansion failed for type :%s slot: %d",
+                                keyType == DB_MAIN ? "main" : "expires", i);
                     return C_ERR;
+                } else {
+                    serverLog(LL_DEBUG, "Dict expansion skipped for type :%s slot: %d",
+                                keyType == DB_MAIN ? "main" : "expires", i);
```

Comment:
why "skipped"?
if it's a try_expand and it failed, we handled it above.
if it was a !try_expand and it failed, then it panicked and killed the process.
so here we know we succeeded.

Reply:
`dictExpand` can return `DICT_ERR` if no expansion happens due to certain conditions. It is not equivalent to the panic/OOM kill.